docs/proposals/policy-feg-api-cleanup/qos_enforcement.md

```diff
@@ -0,0 +1,333 @@
+  vlan_id: string
+  # uplink_bridge must be configured with the uplink interface for this value 
+  # to be valid.
+  uplink_interfaces: string
+  # Prefix for ipv6 allocation
+  # Not yet supported at this time
+  # Usage of this field will depend on PDN Type set
+  ipv6_prefix: string
+```
+
+```
+lte_network:
+  id: string
+  name: string
+  description: string
+  ...
+  apn_resources: array[string]
+  ...
+```
+
+```
+network_epc_configs:
+  ...
+  hss_relay_enabled: bool
+  gx_gy_relay_enabled: bool
+  ...
+```
+The `relay_enabled` flag will be split to enable federation under different
+circumstances.
+
+```
+feg_lte_network:
+  id: string
+  name: string
+  description: string
+  ...
+  apn_resources: array[string]
+  ...
+```
+
+For both LTE and federated LTE networks, APN resources must be associated.
+
+```
+subscriber:
+  id: string
+  name: string
+  ...
+  active_base_names: array[base_name]
+  active_policies: array[policy_id]
+  active_apns: array[string]
+  ...
+```
+
+One or more APN objects will be allowed to be associated to subscribers. 
+This is similar to before, though will now create restrictions on associating 
+policies to subscribers.
+
+As before, one or more policies will be allowed to be associated to 
+subscribers. A new limitation will be created where each policy that is 
+associated to a subscriber must share at least one of its QosProfile’s APNs 
+with the subscriber. This must hold true for each policy to subscriber 
+association, and CRUD operations which could break this rule should return 
+an API error and fail to complete.
+
+```
+policy_rule:
+  id: string
+  flow_list: array[flow_description]
+  priority: uint32
+  rating_group: uint32
+  monitoring_key: string
+  ...
+  qos: string
+```
+
+One QoS objects can be associated to each policy object.
+To implement this, its *qos* field will be changed from the current *flow_qos* 
+type to an ID specifier for the new QoS object type, as defined above.
+
+
+### Pipelined Changes for Enforcement
+
+At a high level, changes in `pipelined` are required to support the following:
+* (P1) Enforce QoS via AMBR across the bearer level, APN level, and subscriber 
+level
+* (P2) Load/KPI reporting (to be used for admission control and session 
+lifecycle management by sessiond)
+* (P3) GBR enforcement
+
+
+### Policydb Changes with Streaming
+
+The `policydb` service across the orc8r and gateway requires changes to support 
+QoS enforcement:
+* (P1) Rename `relay_enabled` flag to `gx_gy_relay_enabled`
+* (P1) Orc8r `policydb` service should be updated to stream to the gateway 
+all the mappings between basenames, policies, QoS profiles, APNs, data plans 
+and subscribers
+* (P1) Gateway `policydb` service should make RAR calls to `sessiond` when 
+updates happen to the policy associations or to policies. To determine when 
+required, policy info and associations are streamed from orc8r, and `policydb` 
+should also query `sessiond` for the relevant info.
```

Comment:
Let's switch to a model where policydb uses a set RPC call to sessiond to update policy rules.